custom-policies/non-prod/pip-frontend.staging.platform.hmcts.net-B2C_1A_TRUSTFRAMEWORKEXTENSIONS_PASSWORD_RESET.xml

```diff
@@ -64,29 +83,74 @@
+          <Protocol Name="Proprietary" Handler="Web.TPEngine.Providers.OneTimePasswordProtocolProvider, Web.TPEngine, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null" />
           <Metadata>
-            <Item Key="Operation">SendCode</Item>
-            <Item Key="allowTestEmailAddressPassThrough">true</Item>
+            <Item Key="Operation">GenerateCode</Item>
+            <Item Key="CodeExpirationInSeconds">1200</Item>
+            <Item Key="CodeLength">6</Item>
+            <Item Key="CharacterSet">0-9</Item>
+            <Item Key="NumRetryAttempts">5</Item>
+            <Item Key="NumCodeGenerationAttempts">10</Item>
+            <Item Key="ReuseSameCode">false</Item>
           </Metadata>
           <InputClaims>
-            <InputClaim ClaimTypeReferenceId="emailAddress" />
+            <InputClaim ClaimTypeReferenceId="emailAddress" PartnerClaimType="identifier" />
           </InputClaims>
+          <OutputClaims>
+            <OutputClaim ClaimTypeReferenceId="verificationCode" PartnerClaimType="otpGenerated" />
+          </OutputClaims>
         </TechnicalProfile>
-        <TechnicalProfile Id="AadSspr-VerifyCode">
+        <TechnicalProfile Id="GenerateBearerAccessToken">
+          <DisplayName>Generate bearer access token</DisplayName>
+          <Protocol Name="Proprietary" Handler="Web.TPEngine.Providers.RestfulProvider, Web.TPEngine, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null" />
+          <Metadata>
+            <Item Key="ServiceUrl">https://login.microsoftonline.com/531ff96d-0ae9-462a-8d2d-bec7c0b42082/oauth2/v2.0/token</Item>
+            <Item Key="AuthenticationType">Basic</Item>
+            <Item Key="SendClaimsIn">Form</Item>
+          </Metadata>
+          <CryptographicKeys>
+            <Key Id="BasicAuthenticationUsername" StorageReferenceId="B2C_1A_SecureRESTClientId" />
+            <Key Id="BasicAuthenticationPassword" StorageReferenceId="B2C_1A_SecureRESTClientSecret" />
+          </CryptographicKeys>
+          <InputClaims>
+            <InputClaim ClaimTypeReferenceId="grant_type" DefaultValue="client_credentials" />
+            <InputClaim ClaimTypeReferenceId="scope" DefaultValue="api://85c405f0-e0a2-4ff1-a16f-e13580d6bea6/.default" />
+          </InputClaims>
+          <OutputClaims>
+            <OutputClaim ClaimTypeReferenceId="bearerToken" PartnerClaimType="access_token" />
+          </OutputClaims>
+          <UseTechnicalProfileForSessionManagement ReferenceId="SM-Noop" />
+        </TechnicalProfile>
+        <TechnicalProfile Id="sendOtpViaPipApim">
+          <DisplayName>REST API call to communicate with pip publication services</DisplayName>
+          <Protocol Name="Proprietary" Handler="Web.TPEngine.Providers.RestfulProvider, Web.TPEngine, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null" />
+          <Metadata>
+            <Item Key="ServiceUrl">https://sds-api-mgmt.test.platform.hmcts.net/pip/publication-services/notify/otp</Item>

```

Comment:
This should be updated to staging before we apply

Reply:
I have updated to the staging url